src/annotator/test/integration/hypothesis-injector-test.js

```diff
@@ -45,217 +42,162 @@ describe('HypothesisInjector integration test', () => {
     // Create a frame before initializing
     const frame = createAnnotatableIFrame();
 
     // Now initialize
     createHypothesisInjector();
+    await onDocumentReady(frame);
 
     // Remove the frame
     frame.remove();
+    await waitForFrameObserver();
 
     assert.calledWith(fakeBridge.call, 'destroyFrame');
   });
 
-  it('injects embed script in frame', () => {
+  it('injects embed script in frame', async () => {
     const frame = createAnnotatableIFrame();
 
     createHypothesisInjector();
-
-    return new Promise(resolve => {
-      isDocumentReady(frame, () => {
-        const scriptElement =
-          frame.contentDocument.querySelector('script[src]');
-        assert(scriptElement, 'expected embed script to be injected');
-        assert.equal(
-          scriptElement.src,
-          config.clientUrl,
-          'unexpected embed script source'
-        );
-        resolve();
-      });
-    });
+    await onDocumentReady(frame);
+
+    const scriptElement = frame.contentDocument.querySelector('script[src]');
+    assert(scriptElement, 'expected embed script to be injected');
+    assert.equal(
+      scriptElement.src,
+      config.clientUrl,
+      'unexpected embed script source'
+    );
   });
 
-  it('excludes injection from already injected frames', () => {
-    const frame = document.createElement('iframe');
-    frame.setAttribute('enable-annotation', '');
-    container.appendChild(frame);
+  it('excludes injection from already injected frames', async () => {
+    const frame = createAnnotatableIFrame();
     frame.contentWindow.eval('window.__hypothesis = {}');
 
     createHypothesisInjector();
+    await onDocumentReady(frame);
 
-    return new Promise(resolve => {
-      isDocumentReady(frame, () => {
-        const scriptElement =
-          frame.contentDocument.querySelector('script[src]');
-        assert.isNull(
-          scriptElement,
-          'expected embed script to not be injected'
-        );
-        resolve();
-      });
-    });
+    assert.isNull(
+      frame.contentDocument.querySelector('script[src]'),
+      'expected embed script to not be injected'
+    );
   });
 
-  it('detects dynamically added frames', () => {
+  it('detects dynamically added frames', async () => {
     // Initialize with no initial frame, unlike before
     createHypothesisInjector();
 
     // Add a frame to the DOM
     const frame = createAnnotatableIFrame();
 
-    return new Promise(resolve => {
-      // Yield to let the DOM and CrossFrame catch up
-      waitForFrameObserver(() => {
-        isDocumentReady(frame, () => {
-          assert(
-            frame.contentDocument.body.hasChildNodes(),
-            'expected dynamically added frame to be modified'
-          );
-          resolve();
-        });
-      });
-    });
+    await waitForFrameObserver();
+    await onDocumentReady(frame);
+    assert.isTrue(
+      frame.contentDocument.body.hasChildNodes(),
```

Comment:
To check my understanding - this test is assuming that the frame initially had an empty `<body>` and that this will change after the client is embedded? This seems brittle to me. I think it would be better to test more directly for one or more of the elements that we expect the client to inject. I'd also recommend encapsulating that test in a helper so you can change it in one place easily.

Reply:
I did not change the test itself, but I will improve it.